salt/metalk8s/addons/prometheus-operator/post-upgrade.sls

```diff
@@ -2,3 +2,9 @@
 
 include:
   - .post-cleanup
+
+Delete old metalk8s-prometheus StorageClass:
+  metalk8s_kubernetes.object_absent:
+    - apiVersion: v1
+    - kind: StorageClass
+    - name: metalk8s-prometheus
```

Comment:
If you remove a storageclass associated with some volumes does the volume still mounted and bound to the Pod (especially after a reboot ?)

I think we may want/need to replace `metalk8s-prometheus` by `metalk8s` on all volumes and all PVC, don't really know how our storage operator will react to that kind of change

Reply:
> If you remove a storageclass associated with some volumes does the volume still mounted and bound to the Pod (especially after a reboot ?)

Good point.

I think we're good, because we don't care about the storage class once the Volume/PV are created (not sure about the ext pillar though).

To confirmed by testing it.

Reply:
The volume will still be bound to the pod even if the storage class no longer exists (even across reboots).
We can change the storage class of a PV, but not a PVC.
We could maybe edit the PV and recreate the PVC, but I'm not sure that something we want to do as it works well without touching them.